src/__tests__/addDefaultProps.coffee

```diff
@@ -0,0 +1,43 @@
+### eslint-disable no-console ###
+import React from 'react'
+import 'jest-dom/extend-expect'
+import {flow} from 'lodash/fp'
+import {render} from 'react-testing-library'
+
+import addDefaultProps from '../addDefaultProps'
+
+Greeting = flow(
+  addDefaultProps ({num}) -> name: if num > 1 then 'People' else 'Person'
+  ({name}) -> <span>Hello, {name}</span>
+)
+
+Greeting2 = flow(
+  addDefaultProps name: 'World'
+  ({name}) -> <span>Hello, {name}</span>
+)
+
+describe 'addDefaultProps', ->
+  test 'adds prop from function when missing', ->
+    {getByText} = render <Greeting num={2} />
+    getByText /Hello,.+People/
+    undefined
+
+  test 'adds prop from object when missing', ->
+    {getByText} = render <Greeting2 />
+    getByText /Hello,.+World/
+    undefined
+
+  test 'adds prop from object when null', ->
+    {getByText} = render <Greeting2 name={null} />
+    getByText /Hello,.+World/
```

Comment:
So I *think* that `react-testing-library` doesn't fully clear the DOM between tests, so I think this test might pass regardless based on the previous test having passed

I'd like to have a clearer mental model of what it's doing/if it can be avoided (since it's typically much easier to assume/reason about there being a fresh DOM for each test), but in the absence of that certainty, I'd recommend using a pattern where you're querying a uniquely identified DOM node in each test

So eg
```
testId = 'nullCheck'
{getByTestId} = <Greeting2 name={null} testId={testId} />
expect(getByTestId testId).toHaveTextContent /Hello,.+World/
```
(and wire up `<Greeting2>` to forward its `testId` prop to `data-testid` of its rendered DOM node)

Reply:
> So I think that react-testing-library doesn't fully clear the DOM between tests, so I think this test might pass regardless based on the previous test having passed

Ugh, you're right, although the docs make it sound like it *should* automatically be cleaning up between tests.